docs/assembly.rst

```diff
@@ -947,11 +980,14 @@ Pseudocode::
       POP
     LabelDefinition(name [id1, ..., idn] :) ->
       JUMPDEST
-      // register new variables id1, ..., idn and set the stack height to
-      // stack_height_at_block_start + number_of_local_variables
+      // register new variables id1, ..., idn and adjust the assumed stack
+      // height such that it matches the stack height at the beginning of
+      // the block plus all local variables including the just registered
+      // ones where idn is at the stack top. If n is zero, resets the stack
+      // height to just the local variables.
```

Comment:
Are there assumed stack contents that correspond to assumed stack height?  Or, is the assumed stack just a height, and the assumed stack does not contain any information about its contents?

Are there some illustrating properties about the assumed stack?